react-native-git-upgrade/cliEntry.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import shell from 'shelljs';
+import Promise from 'promise';
+import yeoman from 'yeoman-environment';
+import TerminalAdapter from 'yeoman-environment/lib/adapter'
+import log from 'npmlog';
+import rimraf from 'rimraf';
+
+import {
+  checkDeclaredVersion,
+  checkMatchingVersions,
+  checkReactPeerDependency,
+  checkGitAvailable,
+  checkNewVersion
+} from './checks';
+
+log.heading = 'git-upgrade';
+
+/**
+ * Promisify the callback-based shelljs function exec
+ * @param command
+ * @param opts
+ * @returns {Promise}
+ */
+function exec(context, command, forceOutput) {
```

Comment:
Hmmm, I don't know, I like the consistency "context-first" of all functions called in the main function:
https://github.com/ncuillery/react-native/blob/react-native-upgrader/react-native-git-upgrade/cliEntry.js#L147-L225

Can we made an exception to this golden rule ?